engine/table/src/main/java/io/deephaven/engine/table/impl/SourceTable.java

```diff
@@ -139,8 +140,7 @@ protected final void initializeAvailableLocations() {
                             new TableLocationSubscriptionBuffer(locationProvider);
                     final TableLocationSubscriptionBuffer.LocationUpdate locationUpdate =
                             locationBuffer.processPending();
-                    throwIfLocationsRemoved(locationUpdate);
-
+                    maybeRemoveLocations(locationUpdate.getPendingRemovedLocationKeys());
```

Comment:
This is confusing. We just made the subscription buffer. It could have removals, I suppose, but none of those locations would have been added to the RCSM yet.

There could be cases where we have the same location as a pending add and a pending remove. I think the correct thing to do is process adds before removes, ironically. Don't forget to apply the same change in the refresh case.

In the long run, we'll need version stamps to tell if we're removing something we actually care about, or not.